ArduPlane/quadplane.cpp

```diff
@@ -562,21 +569,28 @@ bool QuadPlane::setup(void)
         break;
     }
 
-    switch (motor_class) {
-    case AP_Motors::MOTOR_FRAME_TRI:
-        motors = new AP_MotorsTri(plane.scheduler.get_loop_rate_hz(), rc_speed);
-        motors_var_info = AP_MotorsTri::var_info;
-        break;
-    case AP_Motors::MOTOR_FRAME_TAILSITTER:
-        motors = new AP_MotorsTailsitter(plane.scheduler.get_loop_rate_hz(), rc_speed);
-        motors_var_info = AP_MotorsTailsitter::var_info;
-        rotation = ROTATION_PITCH_90;
-        break;
-    default:
+    if (tailsitter.motor_mask == 0) {
         motors = new AP_MotorsMatrix(plane.scheduler.get_loop_rate_hz(), rc_speed);
```

Comment:
I think this should be under default as it was, otherwise you might construct it without deconstructing it when you overwrite it..... or something, not 100% sure.

Reply:
good catch; interesting copy/paste error